masonry_core/src/core/widget.rs

```diff
@@ -134,6 +134,14 @@ pub type ChildrenIds = SmallVec<[WidgetId; 16]>;
 /// widget should only be mutated either during a method call or through a [`WidgetMut`](crate::core::WidgetMut).
 #[allow(unused_variables, reason = "Default impls don't use method arguments")]
 pub trait Widget: AsDynWidget + Any {
+    /// The action type that will widget will submit.
+    ///
+    /// If this widget never submits action, this can be an empty type
+    /// such as [`NoAction`](crate::core::NoAction).
```

Comment:
This probably should have similar security docs as the Xilem Core Message type does. However, this doesn't need to block this PR.